p2p/metrics.go

```diff
@@ -41,12 +41,69 @@ type metricsLabelCache struct {
 	messageLabelNames map[reflect.Type]string
 }
 
+type peerPendingMetricsCache struct {
+	mtx             sync.Mutex
+	perMessageCache map[reflect.Type]peerPendingMetricsCacheEntry
+}
+
+type peerPendingMetricsCacheEntry struct {
+	label            string
+	pendingSendBytes int
+	pendingRecvBytes int
+}
+
+func peerPendingMetricsCacheFromMlc(mlc *metricsLabelCache) *peerPendingMetricsCache {
+	pendingCache := &peerPendingMetricsCache{
+		perMessageCache: make(map[reflect.Type]peerPendingMetricsCacheEntry),
+	}
+	if mlc != nil {
+		mlc.mtx.RLock()
+		for k, v := range mlc.messageLabelNames {
+			pendingCache.perMessageCache[k] = peerPendingMetricsCacheEntry{label: v}
+		}
+		mlc.mtx.RUnlock()
+	}
+	return pendingCache
+}
+
+func (c *peerPendingMetricsCache) AddPendingSendBytes(msgType reflect.Type, addBytes int) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	if entry, ok := c.perMessageCache[msgType]; ok {
+		entry.pendingSendBytes += addBytes
+		c.perMessageCache[msgType] = entry
+	} else {
+		c.perMessageCache[msgType] = peerPendingMetricsCacheEntry{
+			label:            buildLabel(msgType),
+			pendingSendBytes: addBytes,
+		}
+	}
+}
+
+func (c *peerPendingMetricsCache) AddPendingRecvBytes(msgType reflect.Type, addBytes int) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	if entry, ok := c.perMessageCache[msgType]; ok {
+		entry.pendingRecvBytes += addBytes
+		c.perMessageCache[msgType] = entry
```

Comment:
```suggestion
		c.perMessageCache[msgType] = entry
```

https://github.com/cometbft/cometbft/pull/3411/files#r1666530037